core/trace/src/main/scala/org/typelevel/otel4s/trace/Span.scala

```diff
@@ -185,4 +191,11 @@ object Span {
       }
   }
 
+  private[this] val spanKey_ : Key[Any] =
+    Key.newKey[SyncIO, Any].unsafeRunSync()
+  private def spanKey[F[_]]: Key[Span[F]] =
+    spanKey_.asInstanceOf[Key[Span[F]]]
```

Comment:
If Vault fails us, it's right here.  Higher-kinded keys have been a nuisance for http4s, too.  There's a soundness violation if I put a `Span[IO]` and fetch a `Span[EitherT[IO, Lmao, Span[IO]]`.